e2e/compatibility_test.go

```diff
@@ -0,0 +1,110 @@
+			digest = strings.Split(line, "sha256:")[1]
+		}
+	}
+	digest = strings.Trim(digest, " \r\n")
+	info.dockerCmd("tag", digest, tag)
+
+	return "sha256:" + digest
+}
+
+func getBundle(t *testing.T, url string) []byte {
+	client := http.Client{Timeout: time.Minute * 1}
+	res, err := client.Get(url)
+	assert.NilError(t, err)
+
+	body, err := ioutil.ReadAll(res.Body)
+	assert.NilError(t, err)
+
+	return body
+}
+
+func createBundle(path string, bundle []byte, invocDigest string, backendDigest string, frontendDigest string) error {
+	bundle = bytes.ReplaceAll(bundle, []byte("<digest-app-e2e-invoc>"), []byte(invocDigest))
+	bundle = bytes.ReplaceAll(bundle, []byte("<digest-backend>"), []byte(backendDigest))
+	bundle = bytes.ReplaceAll(bundle, []byte("<digest-frontend>"), []byte(frontendDigest))
+
+	return ioutil.WriteFile(path, bundle, 0644)
+
+}
+
+func TestBackwardsCompatibilityV1(t *testing.T) {
+	runWithDindSwarmAndRegistry(t, func(info dindSwarmAndRegistryInfo) {
+		appName := "app-e2e"
+		tmpDir := fs.NewDir(t, appName)
+		defer tmpDir.Remove()
+
+		bundleTemplate := getBundle(t, "https://github.com/docker/app-e2e/raw/master/bundles/v0.9.0/bundle.json")
+
+		invocDigest := importAndTagImage(info, tmpDir, "app-e2e:0.1.0-invoc", "https://github.com/docker/app-e2e/raw/master/images/v0.9.0/app-e2e-invoc.tar")
+		backendDigest := importAndTagImage(info, tmpDir, "app-e2e/backend", "https://github.com/docker/app-e2e/raw/master/images/v0.9.0/backend.tar")
+		frontendDigest := importAndTagImage(info, tmpDir, "app-e2e/frontend", "https://github.com/docker/app-e2e/raw/master/images/v0.9.0/frontend.tar")
+
+		err := createBundle(tmpDir.Join("bundle.json"), bundleTemplate, invocDigest, backendDigest, frontendDigest)
+		assert.NilError(t, err)
+
+		// Install a Docker App built with an older version
+		output := info.dockerCmd("app", "run", "--cnab-bundle-json", tmpDir.Join("bundle.json"), "--name", appName)
+		checkContains(t, output,
+			[]string{
+				fmt.Sprintf("Creating service %s_backend", appName),
+				fmt.Sprintf("Creating service %s_frontend", appName),
+				fmt.Sprintf("Creating network %s_default", appName),
+			})
+		// List the installed application
+		output = info.dockerCmd("app", "ls")
+		checkContains(t, output,
+			[]string{
+				`RUNNING APP\s+APP NAME\s+LAST ACTION\s+RESULT\s+CREATED\s+MODIFIED\s+REFERENCE`,
+				fmt.Sprintf(`%s\s+%s \(0.1.0\)\s+install\s+success\s+.+second[s]?\sago\s+.+second[s]?\sago\s+`, appName, appName),
+			})
+		time.Sleep(time.Second * 10)
+		output = info.execCmd("/usr/bin/wget", "-O", "-", "http://localhost:8080")
+		checkContains(t, output,
+			[]string{`Hi there, I love Docker!`})
+
+		// Update the application, changing the port
+		output = info.dockerCmd("app", "update", appName, "--set", "ports.frontend=8081")
+		checkContains(t, output,
+			[]string{
+				fmt.Sprintf("Updating service %s_backend", appName),
+				fmt.Sprintf("Updating service %s_frontend", appName),
+			})
+		time.Sleep(time.Second * 10)
```

Comment:
I think we should poll until we get the right result instead of sleeping an arbitrary amount of time, to reduce flakiness. See compose on kube: https://github.com/docker/compose-on-kubernetes/blob/master/e2e/compose_test.go#L419